src/menus/EditMenus.cpp

```diff
@@ -615,8 +615,15 @@ void OnPaste(const CommandContext &context)
                   bPastedSomething = true;
                   // For correct remapping of preserved split lines:
                   PasteTimeWarper warper{ t1, t0 + src->GetEndTime() };
-                  wn.ClearAndPaste(t0, t1,
-                     *static_cast<const WaveTrack*>(src), true, true, &warper);
+                  // New desired behaviour as of 3.4: pasting should result in a
+                  // new clip - don't erase boundaries to surrounding clips ...
+                  constexpr auto merge = false;
+                  // ... and of course, don't preserve the boundaries strictly
+                  // in [t0, t1].
+                  constexpr auto preserveExistingBoundaries = false;
+                  wn.ClearAndPaste(
+                     t0, t1, *static_cast<const WaveTrack*>(src),
+                     preserveExistingBoundaries, merge, &warper);
```

Comment:
This commit is small, but I remember now the doubts I had about it.  The changes in Generator led me here, and just that led me to separate the destructive effects branch first, and get all of its problems figured out first.

My doubt is because is that the boolean `merge` governs more than just the internal splitting of the new pasted clip:  see all the rest in `ClearAndPasteOne` under `if (merge && splits.size() > 0)`

I was thinking then, and write it down now, that I think the correct fix may require more boolean arguments to control the details more finely.

Reply:
Contrast the behavior:
- Select part of a track, with a stretch.  Apply an effect. Drag the new clip to another track.  Drag (don't stretch) clip edges.
- Similar but pasting instead of applying an effect (and there doesn't need to be any stretch).

For the effect, the clips containing unchanged data retain hidden clip information.  But for paste -- as a consequence of this commit -- they do not.  (Without this commit, of course there were no new splits at the selection bounds.)

Which is the correct behavior?  I think there should be consistency between these cases.

As for the new clip in the middle -- for the effect, it turns out to preserve the rest of the clip, stretched but without the effect applied.  (But #5220 will change that behavior for performance reasons).

For paste, the middle clip can still carry hidden data from the place from which it was originally cut or copied.

Reply:
Great catch, thank you.
We could add one more boolean, to control the `split` argument of the internal call to `HandleClear`, currently hard-coded to false. This would solve the problem you describe, but would introduce new problems: "There is not enough room ..." if the pasted data is larger than `[t0, t1]`, and a unfilled gap if the pasted data is less than `[t0, t1]`.
And me believing I was done with it ...

Reply:
[Generalize ClearAndPaste to allow preservation of cleared interval](https://github.com/audacity/audacity/pull/5043/commits/fbfbbf5f5d1ab873db6eec0437a6130a0def8f41)